utils/ort/src/test/kotlin/UtilsTest.kt

```diff
@@ -379,4 +387,51 @@ class UtilsTest : WordSpec({
             normalizeVcsUrl(url) shouldBe url
         }
     }
+
+    "requestPasswordAuthentication" should {
+        "return a correct authentication object for a host, port, and scheme combination" {
+            val host = "www.example.org"
+            val port = 442
+            val scheme = "https"
+
+            mockkStatic("org.ossreviewtoolkit.utils.ort.UtilsKt")
+            mockkStatic(Authenticator::class)
+            val proxySelector = mockk<OrtProxySelector>()
+            val passwordAuth = mockk<PasswordAuthentication>()
+
+            every { installAuthenticatorAndProxySelector() } returns proxySelector
+            every {
+                Authenticator.requestPasswordAuthentication(host, null, port, scheme, null, null)
+            } returns passwordAuth
+
+            requestPasswordAuthentication(host, port, scheme) shouldBe passwordAuth
+
+            verify {
+                installAuthenticatorAndProxySelector()
+            }
+        }
+
+        "return a correct authentication object for a URI" {
+            val host = "www.example.org"
+            val port = 442
+            val scheme = "https"
+            val uri = URI(scheme, "$host:$port", "/some/path", "someQuery", "someFragment")
```

Comment:
IMO it would read nicer / fit better to the title of test if this URI was constructed from a string, and line 427 would refer to the URI components like `uri.host` etc.

Reply:
If it helps to get the change in, I am fine with removing these tests. Just want to mention that this unremarkable two-liner function actually has some impact. The implicit assumption that the `OrtAuthenticator` was available caused us a lot of trouble recently.